pkg/client/client.go

```diff
@@ -105,7 +105,7 @@ type client struct {
 }
 
 // resetGroupVersionKind is a helper function to restore and preserve GroupVersionKind on an object.
-// TODO(vincepri): Remove this function and its calls once    controller-runtime dependencies are upgraded to 1.15.
+// TODO(vincepri): Remove this function and its calls once controller-runtime dependencies are upgraded to 1.16?
```

Comment:
Can we drop this function?

Reply:
I thought we could. I removed it in ffe4ab3e610f6f9167f9cc7b52c41502d3b0afef and then reverted that change in the next commit after the tests failed. It seems that 1.15 doesn't fix it.

@vincepri Any clues about where we can look to find the upstream change that would allow us to remove this function?

Reply:
IIRC this is related to apimachinery version. We're at an offsite for the next couple of days so I won't be able to track this down. Are client-go (https://github.com/kubernetes/client-go/blob/master/INSTALL.md#add-client-go-as-a-dependency) and the related dependencies up to date?